### PR TITLE
feat: implement orders history, order details

### PR DESCRIPTION
**Pull request by OlyaAn**
Implement orders history,  order details

## Related Issue
[AB#60287](https://dev.azure.com/orckestra001/da965986-c85b-43be-9915-219568b000e5/_workitems/edit/60287)
[AB#60288](https://dev.azure.com/orckestra001/da965986-c85b-43be-9915-219568b000e5/_workitems/edit/60288)

## Screenshots (if appropriate):
image
image

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: packages/theme/pages/MyAccount/OrderHistory.vue [hunk 1]
```diff
@@ -0,0 +1,466 @@
...
+        case 'InProgress':
+          return 'text-warning';
+        case 'PendingProcess':
+          return 'text-warning';
+        case 'Finalised':
+          return 'text-success';
+        default:
+          return '';
+      }
+    };
```
Comment (InnaBoitsun) [inline packages/theme/pages/MyAccount/OrderHistory.vue:280]: update to have like this:

 ```
const getStatusTextClass = (order) => {
      const status = ordersHistoryGetters.getStatus(order);
      switch (status) {
        case 'InProgress':
          return 'text-warning';
        case 'Completed':
          return 'text-success';
        case 'Canceled':
          return 'text-danger';
        default:
          return 'text-info';
      }
    };
```

File: packages/theme/pages/MyAccount/OrderHistory.vue [hunk 2]
```diff
@@ -0,0 +1,466 @@
...
+      "Creation Date",
+      "Order Total",
+      "Order Status",
+    ];
+
+    const tableHeadersPast = [
+      "Order Number",
+      "Creation Date",
+      "Order Status",
+    ];
```
Comment (InnaBoitsun) [inline packages/theme/pages/MyAccount/OrderHistory.vue:266]: tables should be the same, please create separate component OrdersList.vue and use it something like this
 <SfTab title="My current orders">
<OrdersList :orders="currentOrders"  .../>
</SfTab>
 <SfTab title="My past orders">
  <OrdersList :orders="pastOrders" .../>

For tablle use next columns lables:
      "Order Number",
      "Date",
      "Total",
      "Status",

File: packages/theme/pages/MyAccount/OrderHistory.vue [hunk 3]
```diff
@@ -0,0 +1,466 @@
...
+                <span :class="getStatusTextClass(pastOrder)">{{ getOrderStatusLookup(pastOrder) }}</span>
+              </SfTableData>
+              <SfTableData class="orders__view orders__element--right">
+                <SfButton class="sf-button--text desktop-only" @click="getOrderDetails(pastOrder)">
+                  {{ $t('View details') }}
+                </SfButton>
+              </SfTableData>
+            </SfTableRow>
+          </SfTable>
+          <p>Total orders Past - {{ totalOrdersPast }}</p>
```
Comment (InnaBoitsun) [inline packages/theme/pages/MyAccount/OrderHistory.vue:195]: plase display totals in Tabs label
My current orders (X)
My past orders (X)

File: packages/theme/pages/MyAccount/OrderHistory.vue [hunk 4]
```diff
@@ -0,0 +1,466 @@
...
+        :value="orderGetters.getSubTotal(currentOrder)"
+        class="sf-property--full-width property"
+      />
+      <div v-for="tax in orderGetters.getTaxes(currentOrder)">
+        <SfProperty
+          :name="th.getTranslation(orderGetters.getTaxName(tax))"
+          :value="orderGetters.getTaxTotal(tax)"
+          class="sf-property--full-width property"
+        />
+      </div>
```
Comment (InnaBoitsun) [inline packages/theme/pages/MyAccount/OrderHistory.vue:122]: the Shipping cost is absent in order summary

**Review by InnaBoitsun — CHANGES_REQUESTED** [review]

File: packages/api-client/src/api/orders/getOrders/index.ts [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+  const startingIndex = (page - 1) * maximumItems;
+  const { id: customerId } = parseUserToken(userToken, myAccount.secretPassphrase);
+  if (!customerId) return null;
+
+  const url = new URL(
+    `/api/orders/${scope}/find`,
+    api.url
+  );
+
+  const filters =
```
[the remark this answers is not in the document]
Comment (OlyaAn) [reply packages/api-client/src/api/orders/getOrders/index.ts]: fixed as filterMember, filterValues

File: packages/api-client/src/types.ts [hunk 6]
```diff
@@ -296,7 +303,50 @@ export type FacetSearchCriteria = TODO;
...
+    lastModifiedBy: string,
+    orderNumber: string,
+    orderStatus: string,
+    scopeId: string,
+    source: string,
+    transactionOrderNumber: number
+};
+
+export type OrderItem = {
+    billingCurrency: string,
```
[the remark this answers is not in the document]
Comment (OlyaAn) [reply packages/api-client/src/types.ts:326]: image
image
UserOrder we use for order details and OrderItem we use for orders in the history object. Here totally different models
@InnaBoitsun

File: packages/api-client/src/api/orders/findOrders/index.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+    maximumItems,
+    startingIndex
+  };
+  const body = {
+    customerId,
+    cultureName: locale,
+    includeFulfillmentStates: true,
+    query
+  };
+  console.log(JSON.stringify(body));
```
Comment (InnaBoitsun) [inline packages/api-client/src/api/orders/findOrders/index.ts:54]: remove console.log

File: packages/api-client/src/api/orders/findOrders/index.ts [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+
+  const { data } = await context.client.post(url.href, body);
+  console.log(JSON.stringify(data));
```
Comment (InnaBoitsun) [inline packages/api-client/src/api/orders/findOrders/index.ts:57]: remove console.log

File: packages/composables/src/getters/orderGetters.ts [hunk 9]
```diff
@@ -56,7 +61,115 @@ function getOrdersTotal(orders: any): number {
...
+function getTotal(order: UserOrder): number {
+  return order?.cart?.total;
+}
+
+// eslint-disable-next-line @typescript-eslint/no-unused-vars
+function getSaving(order: UserOrder): number {
+  let saving = 0;
+  order?.cart?.shipments[0]?.lineItems.forEach(item => saving += item?.regularPrice - item?.listPrice);
+  return saving;
+}
```
Comment (InnaBoitsun) [inline packages/composables/src/getters/orderGetters.ts:137]: this getter is wrong, calculation is wrong
it should include Item +  Saving + Coupons + rewards 

for now you can remove using it in Order Details, we will need to fix it later as bug

**Review by InnaBoitsun — APPROVED** [review]

**Pull request merged**